### PR TITLE
Fix meta_data handling

### LINKED ISSUE #37: meta_data attribute
Hi,

I am trying to list builds based on the value of meta_data variables:

```
builds = buildkite.builds().list_all_for_pipeline(org_name,
                                                  pipeline_name,
                                                  meta_data={'origin_id': 0}
                                                  )
```

 but it seems that pybuildkite is building the url as `...?metadata={'origin':0}&...` instead of `...?metadata[origin]=0&...` as defined in Buildkite API documentation.

I checked the unit tests but there doesn't seem to have anything using it.

If this is indeed a bug I would be happy to make a PR to fix it.

### PR DESCRIPTION
**Pull request by Sup3rGeo**
Fixes #37 , includes unit tests

### CODE REVIEW

File: pybuildkite/builds.py [hunk 1]
```diff
@@ -291,6 +294,10 @@ def rebuild_build(self, organization, pipeline, build_number):
             + rebuild
         )
 
+    @staticmethod
+    def __process_meta_data(meta_data):
```
Comment (pyasi) [inline pybuildkite/builds.py]: This method will throw an error if meta_data is `None`. Perhaps we should only call it if meta_data exists. I also think meta_data should be defaulted to be an empty dictionary so it's more clear to the user what type to use.

File: tests/test_builds.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    """
+
+    @pytest.fixture
+    def fake_client(self):
+        """
+        Build a fake API client
+        """
+        return Mock(get=Mock())
+
+    def test_meta_data_url(self, fake_client):
```
Comment (pyasi) [inline tests/test_builds.py]: We'll probably want to add a test where `meta_data` isn't added to verify the bug I listed above no longer exists

File: pybuildkite/builds.py (hunk 1 above)
Comment (Sup3rGeo) [reply pybuildkite/builds.py]: Nice catch thanks! I decided to treat `None` inside this function so it is consistent with the `__get_build_states_query_param(states)` function which is also always called and just returns None if the input is not set:

https://github.com/pyasi/pybuildkite/blob/be821c2f1cac04ea622b72d322f1ee57d937a597/pybuildkite/builds.py#L318-L328

File: tests/test_builds.py (hunk 2 above)
Comment (Sup3rGeo) [reply tests/test_builds.py]: Yup, added! And in fact it would fail :)

**Comment by pyasi** [comment]
This LGTM, can you run `black` though? In particular the new unit test file needs to be reformatted =).

Thanks!

**Comment by Sup3rGeo** [comment]
Done!

**Pull request merged**